desc/compute/utils.py

```diff
@@ -0,0 +1,303 @@
+import os
+
+from desc.backend import jnp
+from desc.grid import LinearGrid, ConcentricGrid
+
+
+def _get_proper_surface(grid, surface_label):
+    """
+    Returns grid quantities associated with the given surface label.
+
+    Parameters
+    ----------
+    grid : Grid, LinearGrid, ConcentricGrid, QuadratureGrid
+        Collocation grid containing the nodes to evaluate at.
+    surface_label : str
+        The surface label of rho, theta, or zeta.
+
+    Returns
+    -------
+    :rtype: (ndarray, ndarray, ndarray)
+    surface_label_nodes : ndarray
+        The column in the grid corresponding to this surface_label's nodes.
+    unique_idx : ndarray
+        The indices of the unique values of the surface_label in grid.nodes.
+    ds : ndarray
+        The differential elements (dtheta * dzeta for rho surface).
+    """
+    if surface_label == "rho":
+        surface_label_nodes = grid.nodes[:, 0]
+        unique_idx = grid.unique_rho_idx
+        ds = grid.spacing[:, 1:].prod(axis=1)
+    elif surface_label == "theta":
+        surface_label_nodes = grid.nodes[:, 1]
+        unique_idx = grid.unique_theta_idx
+        ds = grid.spacing[:, [0, 2]].prod(axis=1)
+    elif surface_label == "zeta":
+        surface_label_nodes = grid.nodes[:, 2]
+        unique_idx = grid.unique_zeta_idx
+        ds = grid.spacing[:, :2].prod(axis=1)
+    else:
+        raise ValueError("Surface label must be 'rho', 'theta', or 'zeta'.")
```

Comment:
might be easier to first just do ``assert surface_label in {"rho", "theta", "zeta"}``